packages/common-all/src/types/configs/publishing/publishing.ts

```diff
@@ -89,6 +92,52 @@ export type GoogleAnalyticsConfig = {
   tracking?: string;
 };
 
+export const publishingSchema = schemaForType<DendronPublishingConfig>()(
+  z
+    .object({
+      enableFMTitle: z.boolean().optional().default(true),
+      // enableHierarchyDisplay?: boolean;
+      // hierarchyDisplayTitle?: string;
+      enableNoteTitleForLink: z.boolean().optional().default(true),
+      enablePrettyRefs: z.boolean().optional().default(true),
+      // enableBackLinks?: boolean;
+      enableKatex: z.boolean().optional().default(true),
+      //
+      // assetsPrefix?: string;
+      copyAssets: z.boolean().default(true),
+      //
+      // canonicalBaseUrl?: string;
+      // customHeaderPath?: string;
+      // ga?: GoogleAnalyticsConfig;
+      // logoPath?: string;
+      // siteFaviconPath?: string;
+      // siteIndex?: string;
+      siteHierarchies: z.array(z.string()).default(["root"]),
+      writeStubs: z.boolean().default(false),
```

Comment:
I had to look that up ;) so it will allow the input to be `undefined` but the output to be non-optional. So it will not throw but return the field (after being parsed) with the default value.

I added this to [[dendron://dendron.docs/pkg.common-all.ref.schemas#optional-vs-default]]